CRM/Member/BAO/Membership.php

```diff
@@ -2178,22 +2173,59 @@ public static function getMembershipContributionId($membershipId, $all = FALSE)
    * IMPORTANT:
    * Sending renewal reminders has been migrated from this job to the Scheduled Reminders function as of 4.3.
    *
+   * @param array $params
+   *   only_active_membership_types, exclude_test_memberships, exclude_membership_status_ids
+   *
    * @return array
    *
    * @throws \CiviCRM_API3_Exception
    * @throws \CRM_Core_Exception
    */
-  public static function updateAllMembershipStatus() {
-    // Tests for this function are in api_v3_JobTest. Please add tests for all updates.
-
-    $updateCount = $processCount = self::updateDeceasedMembersStatuses();
-
+  public static function updateAllMembershipStatus($params) {
+    if (empty($params['only_active_membership_types'])) {
+      $params['only_active_membership_types'] = TRUE;
+    }
+    if (empty($params['exclude_test_memberships'])) {
+      $params['exclude_test_memberships'] = TRUE;
+    }
     // We want all of the statuses as id => name, even the disabled ones (cf.
     // CRM-15475), to identify which are Pending, Deceased, Cancelled, and
     // Expired.
     $allStatus = CRM_Member_BAO_Membership::buildOptions('status_id', 'validate');
+    if (empty($params['exclude_membership_status_ids'])) {
+      $params['exclude_membership_status_ids'] = [
+        array_search('Pending', $allStatus),
+        array_search('Cancelled', $allStatus),
+        array_search('Expired', $allStatus) ?: 0,
+        array_search('Deceased', $allStatus),
+      ];
+    }
+    // Deceased is *always* excluded because it is has very specific processing below.
```

Comment:
I don't see any very specific processing below - am I missing something?

Reply:
@michaelmcandrew This line `$updateCount = $processCount = self::updateDeceasedMembersStatuses();`

Reply:
OK - makes sense now - thanks for the clarification.